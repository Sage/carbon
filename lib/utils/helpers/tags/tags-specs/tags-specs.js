'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var elementsTagTest = function elementsTagTest(wrapper, elements) {
  elements.forEach(function (element) {
    it('include \'data-element="' + element + '"\'', function () {
      expect(wrapper.find({ 'data-element': element }).length).toEqual(1);
    });
  });
};

var rootTagTest = function rootTagTest(rootNode, comp, elem, role) {
  expect(rootNode.prop('data-component')).toEqual(comp);
  expect(rootNode.prop('data-element')).toEqual(elem);
  expect(rootNode.prop('data-role')).toEqual(role);
};

exports.elementsTagTest = elementsTagTest;
exports.rootTagTest = rootTagTest;