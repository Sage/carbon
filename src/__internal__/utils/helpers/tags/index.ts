export { default } from "./tags";
export type { TagProps } from "./tags";
