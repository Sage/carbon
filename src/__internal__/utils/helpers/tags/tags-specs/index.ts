export { elementsTagTest, rootTagTest } from "./tags-specs";
