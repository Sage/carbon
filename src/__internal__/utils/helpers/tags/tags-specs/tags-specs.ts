import { ReactWrapper, ShallowWrapper } from "enzyme";

const elementsTagTest = (wrapper: ReactWrapper, elements: string[]) => {
  elements.forEach((element) => {
    it(`include 'data-element="${element}"'`, () => {
      expect(wrapper.find({ "data-element": element }).length).toEqual(1);
    });
  });
};

const rootTagTest = (
  rootNode: ReactWrapper | ShallowWrapper,
  comp: string,
  elem?: string,
  role?: string
) => {
  expect(rootNode.prop("data-component")).toEqual(comp);
  expect(rootNode.prop("data-element")).toEqual(elem);
  expect(rootNode.prop("data-role")).toEqual(role);
};

// eslint-disable-next-line jest/no-export
export { elementsTagTest, rootTagTest };
