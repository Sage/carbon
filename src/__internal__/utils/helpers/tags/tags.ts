interface DataProps {
  /** Identifier used for testing purposes, applied to the root element of the component. */
  "data-element"?: string;
  /** Identifier used for testing purposes, applied to the root element of the component. */
  "data-role"?: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [restKeys: string]: any;
}

export interface TagProps extends DataProps {
  /** Identifier used for testing purposes, applied to the root element of the component. */
  "data-component"?: string;
}

/**
 * Builds props object containing top level data tags
 */
function tagComponent(
  componentName: string | undefined,
  props: DataProps
): TagProps {
  const tagProps: TagProps = {
    "data-component": componentName,
  };

  if (props["data-element"]) {
    tagProps["data-element"] = props["data-element"] as string;
  }
  if (props["data-role"]) {
    tagProps["data-role"] = props["data-role"] as string;
  }

  return tagProps;
}

export default tagComponent;
