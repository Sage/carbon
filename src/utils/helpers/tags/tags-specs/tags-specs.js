const elementsTagTest = (wrapper, elements) => {
  elements.forEach((element) => {
    it(`include 'data-element="${element}"'`, () => {
      expect(wrapper.find({ 'data-element': element }).length).toEqual(1);
    });
  });
};

const rootTagTest = (rootNode, comp, elem, role) => {
  expect(rootNode.prop('data-component')).toEqual(comp);
  expect(rootNode.prop('data-element')).toEqual(elem);
  expect(rootNode.prop('data-role')).toEqual(role);
};

export {
  elementsTagTest,
  rootTagTest
};
