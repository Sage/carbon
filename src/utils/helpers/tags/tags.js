/**
 * Builds props object containing top level data tags
 *
 * @param {String} name of component
 * @param {Object} component props
 * @return {Object} dataTagProps
 */
function tagComponent(component, props) {
  const tagProps = {
    ['data-component']: component
  };

  if (props['data-element']) { tagProps['data-element'] = props['data-element']; }
  if (props['data-role']) { tagProps['data-role'] = props['data-role']; }

  return tagProps;
}

export {
  tagComponent
};
