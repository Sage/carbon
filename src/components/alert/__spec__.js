import React from 'react';
import TestUtils from 'react/lib/ReactTestUtils';
import Dialog from 'components/dialog'
import Alert from './alert';
import { shallow } from 'enzyme';
import { elementsTagTest, rootTagTest } from '../../utils/helpers/tags/tags-specs';

describe('Alert', () => {
  let instance;
  let onCancel = jasmine.createSpy('cancel');

  beforeEach(() => {
    instance = TestUtils.renderIntoDocument(
      <Alert
        onCancel={ onCancel }
        open={ true }
        title="Alert title" />
    );
  });

  describe('dialogTitleClasses', () => {
    it('return the dialog title class along with the alert title class', () => {
      expect(instance.dialogTitleClasses).toEqual('carbon-dialog__title carbon-alert__title');
    });
  });

  describe('dialogClasses', () => {
    it('returns the dialog class along with the alert class', () => {
      expect(instance.dialogClasses).toEqual('carbon-dialog__dialog carbon-dialog__dialog--extra-small carbon-alert__alert');
    });
  });

  describe("tags", () => {
    describe("on component", () => {
      let wrapper = shallow(<Alert open={ true } data-element='bar' data-role='baz' />);

      it('include correct component, element and role data tags', () => {
        rootTagTest(wrapper, 'alert', 'bar', 'baz');
      });
    });

    describe("on internal elements", () => {
      let wrapper = shallow(<Alert open={ true } title='Test' subtitle='Test' showCloseIcon={ true } />);

      elementsTagTest(wrapper, [
        'close',
        'subtitle',
        'title'
      ]);
    });
  });
});
