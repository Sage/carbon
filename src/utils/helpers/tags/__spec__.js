import { tagComponent } from './tags';

describe('tagComponent', () => {
  describe('when no additional tag props are sent', () => {
    it('returns the component name tag', () => {
      expect(tagComponent('my-component', {})).toEqual({ ['data-component']: 'my-component' });
    });
  });

  describe('when role and element props are sent', () => {
    it('adds those to the tagProps object', () => {
      expect(tagComponent('my-component', { ['data-element']: 'my-component', ['data-role']: 'contacts'}))
        .toEqual(
          {
            ['data-component']: 'my-component',
            ['data-element']: 'my-component',
            ['data-role']: 'contacts'
          }
        );
    });
  });
});
