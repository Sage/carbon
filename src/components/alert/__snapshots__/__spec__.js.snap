// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert include correct component, element and role data tags 1`] = `
<Portal
  key="1"
>
  <div
    className="carbon-dialog"
    data-component="alert"
    data-element="bar"
    data-role="baz"
    data-state="open"
  >
    <CSSTransitionGroup
      component="div"
      transitionAppear={true}
      transitionAppearTimeout={500}
      transitionEnter={true}
      transitionEnterTimeout={500}
      transitionLeave={true}
      transitionLeaveTimeout={500}
      transitionName="modal-background"
    >
      <div
        className="carbon-modal__background"
      />
    </CSSTransitionGroup>
    <CSSTransitionGroup
      component="div"
      transitionAppear={true}
      transitionAppearTimeout={500}
      transitionEnter={true}
      transitionEnterTimeout={500}
      transitionLeave={true}
      transitionLeaveTimeout={500}
      transitionName="modal"
    >
      <div
        aria-describedby="carbon-dialog-subtitle"
        aria-labelledby="carbon-dialog-title"
        className="carbon-dialog__dialog carbon-dialog__dialog--extra-small carbon-alert__alert"
        data-component="alert"
        data-element="bar"
        data-role="baz"
        onBlur={[Function]}
        role="dialog"
        style={
          Object {
            "minHeight": undefined,
          }
        }
        tabIndex={0}
      >
        <div
          className="carbon-dialog__title carbon-dialog__title--has-subheader"
        >
          <Heading
            divider={true}
            separator={false}
            subheader="Alert Subtitle"
            subtitleId="carbon-dialog-subtitle"
            title="Alert title"
            titleId="carbon-dialog-title"
          />
        </div>
        <div
          className="carbon-dialog__content"
        >
          <div
            className="carbon-dialog__inner-content"
            style={Object {}}
          />
        </div>
        <Icon
          bgSize="small"
          className="carbon-dialog__close"
          data-element="close"
          onBlur={[Function]}
          onClick={[Function]}
          tabIndex="0"
          type="close"
        />
      </div>
    </CSSTransitionGroup>
  </div>
</Portal>
`;
